network/framework/src/peer.rs

```diff
@@ -0,0 +1,816 @@
+                                let queue_micros = (std::time::SystemTime::now().duration_since(std::time::UNIX_EPOCH).unwrap().as_micros() as u64) - enqueue_micros;
+                                counters::network_peer_outbound_queue_time(self.role_type.as_str(), self.peer_network_id.network_id().as_str(), msg.protocol_id_as_str(), queue_micros);
+                                if estimate_serialized_length(&msg) > self.max_frame_size {
+                                    // start stream
+                                    self.start_large(msg)
+                                } else {
+                                    MultiplexMessage::Message(msg)
+                                }
+                            },
+                        },
+                        // TODO: why does select on close.wait() work below but I did this workaround here?
+                        _wait_result = closed.done.wait_for(|x| *x) => {
+                            close_reason = "closed done 1";
+                            break;
+                        },
+                    }
+                }
+            };
+            if let MultiplexMessage::Message(NetworkMessage::Error(ErrorCode::DisconnectCommand)) = &mm {
+                // some app code decided we should disconnect from this peer
+                close_reason = "got DisconnectCommand";
+                counters::connection_closed(self.role_type.as_str(), self.peer_network_id.network_id().as_str(), "dc");
+                break;
+            }
+            let data_len = mm.data_len();
+            tokio::select! {
+                send_result = self.writer.send(&mm) => match send_result {
+                    Ok(_) => {
+                        // info!("writer_thread ok sent {:?} bytes", data_len);
+                        peer_message_frames_written(&self.peer_network_id.network_id()).inc();
+                        peer_message_bytes_written(&self.peer_network_id.network_id()).inc_by(data_len as u64);
+                    }
+                    Err(err) => {
+                        // TODO: counter net write err
+                        close_reason = "send error";
+                        warn!("writer_thread error sending [{:?}]message to peer: {:?} mm={:?}", data_len, err, mm);
+                        break;
+                    }
+                },
+                _ = closed.wait() => {
+                    close_reason = "closed done 2";
+                    break;
+                }
+            }
+        }
+        info!(
+            reason = close_reason,
+            op = "writer_thread closing",
+            peer = self.peer_network_id,
+            "peerclose"
+        );
+        closed.close().await;
+    }
+}
+
+pub static NETWORK_PEER_MESSAGE_FRAMES_WRITTEN: Lazy<IntCounterVec> = Lazy::new(||
```

Comment:
Do we want to keep these counters in this file, or move them to the dedicated file? Probably makes sense to be consistent and either use a single file, or move all the counters to appropriate places.